### DISCUSSION TITLE
Does Microsoft have MAUI production apps?

### DISCUSSION THREAD

**Discussion by AncientLust** [dotnet/maui#20202] [General] [answered]
Another day of MAUI development. Another "try to find the root issue" day, another bug report. And I just had a thought. 

Does Microsoft have production MAUI applications? I mean if no, why developers should spend their time fighting all these issues if Microsoft cannot rely on their own framework? I saw MAUI showcase, but that’s just a bunch of toys made in 3-5 days. 

Can someone share an example of MAUI app which you can literally download from App Store/Play Market? If no, other question here is why Microsoft doesn't try to migrate their products to MAUI? I know another 1000-2000 bug reports will be raised, but at least they will be solved at a high priority and MAUI will have a production ready showcase app.

Have to go, today I found that app crashes on one page in release build and works fine in debug build. Never ending story.

**Comment by Yannikk1996** [comment]
We have moved from Xamarin to Maui and have now 4 Apps ready for release. We had to rewrite most controls and used our own implementation. Started with .NET 7 and needed a lot of workarounds but it was ok because we were able to use our existing backend for bluetooth,nfc, server connection, camera, microphone etc. Now with .NET 8 and SR1 the critical stuff is fixed. All apps are available for iOS, Android and Windows. 

One of our apps is vision control:

iOS: https://apps.apple.com/de/app/vision-control/id6467643102
Android: https://play.google.com/store/apps/details?id=de.kkt.Vision&gl=DE
Windows: https://apps.microsoft.com/detail/9PN2FRZHH142?hl=de-de&gl=DE

Unfortunately the app can‘t be used without external bluetooth hardware.

**Reply by akhanalcs** [reply]
Your apps look nice. Great job!

**Reply by azrinsani** [reply]
How was your experience building the app?

**Reply by Yannikk1996** [reply]
The experience starting with .NET 7 was disappointing. We have done so much workarounds that required an unbelievable amount of time. Also Tooling with Visual Studio was terrible. After we stopped using controls like collectionview, carouselviews and even shell and Titleview, it began working. All more complex elements had many bugs at this time. Now we do not use much more than standard layouts (like Grid, Stacklayout), Scrollview, Images, Buttons and Borders. The more complex stuff is done via Skiasharp, which works quite well. With the update to .NET 8 most bugs of the simple controls were gone and the performance improved very much. Now with SR1 the experience using simple controls is really good and every release feels better.

**Reply by Xyncgas** [reply]
In terms of man hours how many did it take for the app

**Comment by akhanalcs** [comment]
MAUI will be great only when MSFT dogfoods it in their flagship apps. Use MAUI in some office apps or Outlook app, not RN.
Without dogfooding, it's hard for developers to have confidence in the product.

**Reply by mikkpaasuke** [reply]
It's irrational and unreasonable to expect MS to rewrite flagship applications that have who-knows how many years of person effort and dependencies wrapped up in them. And .NET MAUI doesn't need to be 'great', it just needs to be 'workable', which in my experience it now is.

**Reply by jho123456** [reply]
"It's irrational and unreasonable to expect MS to rewrite flagship applications that have who-knows how many years of person effort and 
dependencies wrapped up in them."

Are you being sarcastic or are you for real? Isn't that what they are expecting Xamarin developers to do?

**Reply by Pastajello** [reply]
Xamarin developers do not rewrite js to c# but Xamarin views and some VMs to MAUI which is mostly the same.

**Comment by pictos** [comment, marked as answer]
Yes, MS has apps made with .NET MAUI, as you can see [here](https://youtu.be/4saU9BNY6l4?t=251)

**Comment by skarnes20** [comment]
A few years ago, I watched a presentation from Microsoft where they discussed their journey with TFS (now Azure DevOps). They highlighted how much the tool had improved once they started using it in their own projects. 

I believe that if they had applied the same approach to cross-platform development, the quality and functionality of MAUI would have seen significant enhancements and trust in MAUI would have grown.

Has there been any progress in the past year? 
Are there more examples of production apps from Microsoft built on MAUI?
